src/main/java/apoc/meta/Meta.java

```diff
@@ -167,7 +168,17 @@ public MetaResult other(List<String> labels) {
         }
 
         public MetaResult type(String type) {
-            this.type = type;
+            if (type == null || type.equals(this.type)) {
+                return this;
+            }
+            if (!this.types.contains(type)) {
+                this.types.add(type);
+            }
+            if (!type.equals(this.type) && this.type != null) {
+                this.type = Types.STRING.toString();
```

Comment:
perhaps it's better to set it null then?
like "undefined" if we use STRING we also need to add STRING to the list?
something else that might make sense if we have INTEGER and FLOAT to coerce to FLOAT

Reply:
But what if we have a field which is a `timestamp` for some values and `localdate` for others? Maybe we can keep the current behaviour as-is as `deprecation` and highlight that in the docs